addons/hr_expense/models/hr_expense.py

```diff
@@ -166,44 +166,78 @@ def _compute_state(self):
     @api.depends('quantity', 'unit_amount', 'tax_ids', 'currency_id')
     def _compute_amount(self):
         for expense in self:
-            if expense.product_id and not expense.unit_amount:
+            if expense.product_id and not expense.product_has_cost:
                 continue
-            taxes = expense._get_taxes(price=expense.unit_amount, quantity=expense.quantity)
-            expense.total_amount = taxes['total_included']
+            base_lines = [expense._convert_to_tax_base_line_dict(price_unit=expense.unit_amount, quantity=expense.quantity)]
+            taxes_totals = self.env['account.tax']._compute_taxes(base_lines, None, True, True)['totals'][expense.currency_id]
+            expense.total_amount = taxes_totals['amount_untaxed'] + taxes_totals['amount_tax']
 
     @api.depends('total_amount', 'tax_ids', 'currency_id')
     def _compute_amount_tax(self):
-        """Note: as total_amount can be set directly by the user (for product without cost) or needs to be computed (for product with cost),
-           `untaxed_amount` can't be computed in the same method as `total_amount`.
+        """
+         Note: as total_amount can be set directly by the user (for product without cost) or needs to be computed (for product with cost),
+        `untaxed_amount` can't be computed in the same method as `total_amount`.
         """
         for expense in self:
-            taxes = expense._get_taxes(price=expense.total_amount, quantity=1.0)
-            expense.amount_tax = taxes['total_included'] - taxes['total_excluded'] if expense.tax_ids else 0.0
-            expense.untaxed_amount = taxes['total_excluded']
+            base_lines = [expense._convert_to_tax_base_line_dict(price_unit=expense.total_amount)]
+            taxes_totals = self.env['account.tax']._compute_taxes(base_lines, None, True, True)['totals'][expense.currency_id]
```

Comment:
```suggestion
            taxes_totals = self.env['account.tax']._compute_taxes(base_lines)['totals'][expense.currency_id]
```
I don't understand why you don't set only the optional parameters you want instead of putting all of them without the kwargs... You should specify the keywords everytime